tsl/src/nodes/decompress_chunk/compressed_batch.h

```diff
@@ -10,29 +10,41 @@
 
 typedef struct ArrowArray ArrowArray;
 
-typedef struct CompressedColumnValues
+/* How to obtain the decompressed datum for individual row. */
+typedef enum
 {
-	/* For row-by-row decompression. */
-	DecompressionIterator *iterator;
-
+	DT_Default = -2,
+	DT_Iterator = -1,
+	DT_Invalid = 0,
 	/*
-	 * For bulk decompression and vectorized filters, mutually exclusive
-	 * with the above.
+	 * Any positive number is also valid for the decompression type. It means
+	 * arrow array of a fixed-size by-value type, with size given by the number.
 	 */
-	ArrowArray *arrow;
+} DecompressionType;
```

Comment:
This will later have two new options for arrow arrays for text columns, with dictionary encoding and without.